src/views/virtualmachinesinstance/details/tabs/disks/hooks/useDisksTableDisks.ts

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react';
+
+import { V1VirtualMachineInstance } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { K8sResourceCommon, useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+
+import {
+  DiskPresentation,
+  diskStructureCreator,
+} from './../utils/virtualMachinesInstancePageDisksTabUtils';
+
+type UseDisksTableDisks = (vmi: V1VirtualMachineInstance) => [DiskPresentation[], boolean, any];
+
+const useDisksTableDisks: UseDisksTableDisks = (vmi) => {
+  const vmiDisks = vmi?.spec?.domain?.devices?.disks;
+  const vmiVolumes = vmi?.spec?.volumes;
+
+  const [pvcs, loaded, loadingError] = useK8sWatchResource<K8sResourceCommon[]>({
+    kind: 'PersistentVolumeClaim',
+    isList: true,
+    namespaced: true,
+    namespace: vmi?.metadata?.namespace,
+  });
+
+  const disks = React.useMemo(() => {
+    const diskDevices = vmiVolumes?.map((volume) => {
+      const disk = vmiDisks?.find(({ name }) => name === volume.name);
+      const pvc = pvcs?.find(
+        ({ metadata }) =>
+          metadata.name === volume?.persistentVolumeClaim?.claimName ||
+          metadata.name === volume?.dataVolume?.name,
+      );
+      return { ...pvc, ...disk };
```

Comment:
why are we merging the fields of pvc and disk together? sounds confusing to me.
why not
```
disk: {
...diskproperties,
pvc
}
```

Reply:
this is used for internal use and consumed later just to create the DiskPresention. no need for speration